File: tutorial/components.py
import dash_core_components as dcc
import dash_html_components as html

from tutorial import styles


def Syntax(children, style=styles.code_container, summary=''):
    code = children
    if not code.startswith('```'):
        code = '```py\n' + code + '\n```'

    if summary:
        return html.Details([
            html.Summary(summary),
            dcc.Markdown(code, style=style)
        ], open=True)
    else:
        return dcc.Markdown(code, style=style)


def Example(example):
    return html.Div(example, className='example-container')

File: dashr/components.R
source('dashr/styles.R')

# Added Reuseable Components

Header <- function(title) {
  return (
    htmlDiv(
      htmlH2(
        list(htmlDiv(title)),
        style = list('borderBottom' = 'thin lightgrey solid', 'marginRight' = 20)
      ),
    )
  )
}

# Existing Components (may need to edit)

Syntax <- function(children, language='R', style=styles$code_container, summary=''){
  if (!(summary == '')){
  return(htmlDetails(list(
    htmlSummary(summary),
    htmlDiv(children)
    ),
    open=TRUE))}
else{
  return(dccSyntaxHighlighter(
    children,
    'language'=language,
    'customStyle'=style
  ))
}}

Example <- function(example){
  return(htmlDiv(example, className='example-container'))}

